optaplanner-core/src/test/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhaseTest.java

```diff
@@ -166,9 +165,8 @@ public void solvingStarted(SolverScope<TestdataSolution> solverScope) {
         assertThat(solutionFuture.get()).isNotNull();
     }
 
-    @Disabled("PLANNER-2249")
     @Test
-    @Timeout(5)
+    @Timeout(15)
```

Comment:
15 of what? I believe the annotation now has a way of specifying the unit.

Reply:
Timeout 15 seconds doesn't mean the test runs for 15 seconds. I'm not sure if the timeouts are necessary at all but I think that if there is a timeout it should be orders of magnitude higher than the regular observed test execution time. For example, if the test is observed to run for 100ms on development machine I wouldn't hesitate to give it a 60 seconds timeout. Why? Because:
1. Under normal conditions it doesn't matter. The test will run for 100ms.
2. If you see a 100ms test hang for 60s it's a clear evidence that there's a race condition in the tested code or in the test itself causing a deadlock. If this happens, timeout increase is out of question. The test needs to be investigated and fixed (or removed if the bug is in the test and there is no reasonable way to make the test stable).
3. Some might argue a 60 seconds timeout is still better than nothing because it avoids the CI job to hang in the broken test until it's aborted. With the timeout, you'll see the test failure in the test report and at the same time, everything after the test will still execute normally.

Reply:
Originally, a timeout of 5 seconds was added to the `shutdownMainThreadAbruptly()` test: https://github.com/kiegroup/optaplanner/pull/340/files and the test appeared to be stable for several years.

Judging by the fact that this test wasn't `@Disabled`, it's still stable and the timeout increase doesn't hurt but what's the point @michaltomco?

Reply:
It has been disabled 11/11/20. Since this PR suppose to stabilize the test I assume it was successfully launched. In that case, I suppose we do not need to stabilize the test, but create a ticket

Reply:
> Timeout 15 seconds doesn't mean the test runs for 15 seconds. I'm not sure if the timeouts are necessary at all but I think that if there is a timeout it should be orders of magnitude higher than the regular observed test execution time. For example, if the test is observed to run for 100ms on development machine I wouldn't hesitate to give it a 60 seconds timeout. Why? Because:
> 
>     1. Under normal conditions it doesn't matter. The test will run for 100ms.
> 
>     2. If you see a 100ms test hang for 60s it's a clear evidence that there's a race condition in the tested code or in the test itself causing a deadlock. If this happens, timeout increase is out of question. The test needs to be investigated and fixed (or removed if the bug is in the test and there is no reasonable way to make the test stable).
> 
>     3. Some might argue a 60 seconds timeout is still better than nothing because it avoids the CI job to hang in the broken test until it's aborted. With the timeout, you'll see the test failure in the test report and at the same time, everything after the test will still execute normally.

Very clear explanation of how we do it. We have very fast tests (milliseconds), but very high timeouts (often 10 minutes), to avoid infinite builds and to avoid random failures on slow machines.